bot/feedback/feedback.py

```diff
@@ -0,0 +1,303 @@
+
+        Allows the owner to mark a suggestion as implemented  and add an optional reason to it. If this is the case, the
+        corresponding embed will be adapted to represent this change and the user who submitted the idea will be
+        notified via DM.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            suggestion_id (int): The id of the suggestion which should be approved.
+            reason (Optional[str]): The reason for the decision.
+        """
+        await ctx.message.delete()
+        await self._change_suggestion_status(SuggestionStatus.IMPLEMENTED, ctx.author, ctx.guild, suggestion_id, reason)
+
+        log.info("Suggestion #%s marked as implemented by %s.", suggestion_id, ctx.author)
+
+    @suggestion_approve.error
+    @suggestion_deny.error
+    @suggestion_consider.error
+    @suggestion_implemented.error
+    async def suggestion_error(self, ctx: commands.Context, error: commands.CommandError):
+        """Error Handler for the `suggestion` subcommands `approve`, `deny`, `consider` and `implemented`.
+
+        Handles specific exceptions which occur during the execution of this command. The global error handler will
+        still be called for every error thrown.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            error (commands.CommandError): The error raised during the execution of the command.
+        """
+        if isinstance(error, commands.BadArgument):
+            await ctx.send("Ich konnte leider keinen Vorschlag mit der von dir angegebenen Nummer finden. :frowning2:",
+                           delete_after=constants.TIMEOUT_INFORMATION)
+
+    async def _change_suggestion_status(self, status: SuggestionStatus, author: discord.Member, guild: discord.Guild,
+                                        suggestion_id: int, reason: Optional[str]):
+        """Method which changes the status of a suggestion.
+
+        Changes the status of the suggestion in our db, adapts the corresponding embed in the suggestion channel to
+        represent the newly made changes and finally notifies the user who submitted it via DM.
+
+        Args:
+            status (SuggestionStatus): The new status of the suggestion.
+            author (discord.Member): The user who invoked the command to change it.
+            guild (discord.Guild): The Discord server on which to find the suggestion.
+            suggestion_id (int): The id of the suggestion.
+            reason (Optional[str]): The reason for the decision.
+        """
+        id_exists = self._db_connector.set_suggestion_status(suggestion_id, status)
+        if not id_exists:
+            raise commands.BadArgument
+
+        suggestion_data = self._db_connector.get_suggestion(suggestion_id)
+        message = await guild.get_channel(constants.CHANNEL_ID_SUGGESTIONS).fetch_message(suggestion_data[0])
```

Comment:
What happens if the msg has been deleted? Did we ever consider this before? Is this a problem?

Reply:
It depends. `fetch_message()` should return `None` which then results in an exception like `NoneType doesn't have method xxxx`. This will be logged but the bot shouldn't be interrupted or even crash because of something like that.